eax/src/traits.rs

```diff
@@ -0,0 +1,43 @@
+use aead::consts::{U1, U10, U11, U12, U13, U14, U15, U16, U2, U3, U4, U5, U6, U7, U8, U9};
+use aead::generic_array::typenum::Unsigned;
+
+mod private {
+    // Sealed traits stop other crates from implementing any traits that use it.
+    pub trait SealedTag {}
+
+    impl SealedTag for super::U1 {}
+    impl SealedTag for super::U2 {}
+    impl SealedTag for super::U3 {}
+    impl SealedTag for super::U4 {}
+    impl SealedTag for super::U5 {}
+    impl SealedTag for super::U6 {}
+    impl SealedTag for super::U7 {}
+    impl SealedTag for super::U8 {}
+    impl SealedTag for super::U9 {}
+    impl SealedTag for super::U10 {}
+    impl SealedTag for super::U11 {}
+    impl SealedTag for super::U12 {}
+    impl SealedTag for super::U13 {}
+    impl SealedTag for super::U14 {}
+    impl SealedTag for super::U15 {}
+    impl SealedTag for super::U16 {}
+}
+
+pub trait TagSize: Unsigned + private::SealedTag {}
+
+impl TagSize for U1 {}
+impl TagSize for U2 {}
+impl TagSize for U3 {}
```

Comment:
Are we sure we want to allow such short tags? Also I think it should be possible to specify a range bound using `typenum` without introducing the sealed trait (see [`typenum::type_operators`](https://docs.rs/typenum/1.12.0/typenum/type_operators/index.html)), though the current approach is simple and easier to understand, so we can leave it as-is.